ddtrace/contrib/pytest/plugin.py

```diff
@@ -306,111 +310,132 @@ def _get_test_class_hierarchy(item):
+                if test_class_hierarchy:
+                    span.set_tag_str(test.CLASS_HIERARCHY, test_class_hierarchy)
+                if hasattr(item, "dtest") and isinstance(item.dtest, DocTest):
+                    span.set_tag_str(test.SUITE, "{}.py".format(item.dtest.globs["__name__"]))
+                else:
+                    span.set_tag_str(test.SUITE, test_suite_span.get_tag(test.SUITE))
+
+            span.set_tag_str(test.TYPE, SpanTypes.TEST)
+            span.set_tag_str(test.FRAMEWORK_VERSION, pytest.__version__)
+
+            if item.location and item.location[0]:
+                _CIVisibility.set_codeowners_of(item.location[0], span=span)
+
+            # We preemptively set FAIL as a status, because if pytest_runtest_makereport is not called
+            # (where the actual test status is set), it means there was a pytest error
+            span.set_tag_str(test.STATUS, test.Status.FAIL.value)
+
+            # Parameterized test cases will have a `callspec` attribute attached to the pytest Item object.
+            # Pytest docs: https://docs.pytest.org/en/6.2.x/reference.html#pytest.Function
+            if getattr(item, "callspec", None):
+                parameters = {"arguments": {}, "metadata": {}}  # type: Dict[str, Dict[str, str]]
+                for param_name, param_val in item.callspec.params.items():
+                    try:
+                        parameters["arguments"][param_name] = encode_test_parameter(param_val)
+                    except Exception:
+                        parameters["arguments"][param_name] = "Could not encode"
+                        log.warning("Failed to encode %r", param_name, exc_info=True)
+                span.set_tag_str(test.PARAMETERS, json.dumps(parameters))
+
+            markers = [marker.kwargs for marker in item.iter_markers(name="dd_tags")]
+            for tags in markers:
+                span.set_tags(tags)
+            _store_span(item, span)
+
+            # Run the actual test
             yield
 
-    nextitem_pytest_module_item = _find_pytest_item(nextitem, pytest.Module)
-    if test_suite_span is not None and (
-        nextitem is None or nextitem_pytest_module_item != pytest_module_item and not test_suite_span.finished
-    ):
-        _mark_test_status(pytest_module_item, test_suite_span)
-        test_suite_span.finish()
-
-    nextitem_pytest_package_item = _find_pytest_item(nextitem, pytest.Package)
-    if test_module_span is not None and (
-        nextitem is None or nextitem_pytest_package_item != pytest_package_item and not test_module_span.finished
-    ):
-        _mark_test_status(pytest_package_item, test_module_span)
-        test_module_span.finish()
+        nextitem_pytest_module_item = _find_pytest_item(nextitem, pytest.Module)
+        if test_suite_span is not None and (
+            nextitem is None or nextitem_pytest_module_item != pytest_module_item and not test_suite_span.finished
```

Comment:
Maybe `nextitem is None` is redundant, because then it should be guaranteed that `nextitem_pytest_module_item != pytest_module_item`.

Reply:
Definitely, removing `nextitem is None` doesn't break any test, so it's worth taking a deeper look there, but I consider these changes outside of the scope of this PR (I'm just indenting the code)